### DISCUSSION TITLE
Can't access to Hubs Enterprise Multi-Server with Domain Recipe #3

### DISCUSSION THREAD

**Discussion by jesus-hergueta** [mozilla/hubs#3028] [Hubs Cloud Questions] [answered]
Hi there

**Description**
I'm creating a new AWS account to host Mozilla Hubs Enterprise Multi-Server and actually all the steps in the Quik Guide worked as well the video tutorials and the guide says.

The problem is that I'm using a 3rd party bought it domain and I can't access to the platform even when the deploy finish successfully. My browser says: DNS_PROBE_FINISHED_NXDOMAIN. I've already tried with “https: //” and “http: //” and using “www.mydomain.com” or “mydomain.com”.
This are my 3rd party DNS Records:
![dns-records](https://user-images.githubusercontent.com/29721843/93172702-59acae00-f6f9-11ea-961f-c548a9d7e199.png)

It seems like the A record with the IP address is missing but where can I find it? I tried with an Elastic IP assigned to the EC2 instance and it works, but I start to have problems with the certificate.

**To Reproduce**
Steps to reproduce the behavior (all AWS settings are in us-east-1):
1. Buy the internal domains in Route 53 as the Recipe # 3 says: mydomain-internal.com and mydomain.link.
2. Create the Certificate in AWS for the 3rd party bought it domain, then wait for the verification.
3. Delete all the DNS records but the Certificate Validation Record in the 3rd Party DNS Server.
4. Delete all the DNS records but the SOA and NS records of both domains in Route 53.
5. Deploy the stack following all the Quick Guide steps for Hubs Enterprise Multi-Server with Domain Recipe # 3.
6. Add to the 3rd party DNS records the CNAME record with the AddressForRootDomain output value.

**Expected behavior**
Can access to the site.

**Additional context**
I can't migrate my domain yet because I bought it 2 weeks ago, but I'm looking forward to migrate it when the 60 days rules ends.

Thank you.

**Comment by robinkwilson** [comment, marked as answer]
You may have used Recipe 3 incorrectly (if that's the case, we need to update the documentation). 
Recipe 3 is only if you're using the root domain for something else in another registrar. For example our hubs.mozilla.com setup, mozilla.com is already used and in a different place other than aws so Recipe 3 is used to setup hubs.mozilla.com.

If you ONLY want to use the ROOT domain for hubs and simply purchased your domain on an external registrar (like Namecheap or GoDaddy), you can follow the guide [Not Setup on Route 53 and Want to be?](https://hubs.mozilla.com/docs/hubs-cloud-aws-domain-recipes.html#is-my-domain-set-up-on-route-53) then you can follow Domain Recipe 1 or 2.

You're not the first to interpret the guide that way, it's an indication that page isn't clear enough yet. 

Let me know how this goes!

**Reply by jesus-hergueta** [reply]
Yes, well I want to use a domain and external DNS that is on a service like GoDaddy.

So should I follow Recipe 1 or 2?

Should I create the domain on Route 53 as a hosted zone or something?

**Reply by jesus-hergueta** [reply]
So, I tried with the Recipe # 2 and the deploy fails:
![Screenshot from 2020-09-15 12-12-13](https://user-images.githubusercontent.com/29721843/93237233-d1a4c380-f74d-11ea-853c-24d12b08cc52.png)

I'm not sure if I should try with number 1 because I cant set mydomain.com in the Internal Domain field since this is a autofill list wich comes from Route 53 and mydomain.com is not there.

**Reply by robinkwilson** [reply]
Did you setup your domain on Route 53 with a hosted zone? If you did and it finished registering, it will show up in the drop-down list, though I personally recommend Recipe 2.

To find the first error message in a Rollback: https://hubs.mozilla.com/docs/hubs-cloud-aws-troubleshooting.html#my-aws-stack-says-rollback-complete-after-deploying-what-went-wrong

**Reply by jesus-hergueta** [reply]
Hi

It works!!! 

I migrate the DNS server of my domain.

1. Delete the all the records and domain configuration in my old DNS server provider.
2. Create the new Hosted Zone in Route 53 so I can get the new nameservers.
3. Setup the new nameservers as NS records in my domain provider service.
4. Follow the Recipe 2 with this settings:


Site 				Domain Name | mydomain.com
-- | --
Site 				is Set Up On Route 53 | Yes
Internal 				Domain | mydomain-internal.com*
Short 				Link Domain | md.link*
Outgoing 				Email Domain | mydomain.com
Outgoing 				Email Subdomain Prefix | mail

_* this domains was purchased directly in AWS._

And then the deploy run smooth, the platform starts without any additional issue or configuration.


Thanks a lot.

Hope this can help someone else too.

**Reply by robinkwilson** [reply]
Amazing! Glad the Hosted Zone approach worked for you!

I'm sure it will help someone else!
